test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/util.py

```diff
@@ -346,3 +357,51 @@ def create_and_invoke_endpoint(
             return invoke_endpoint(
                 sagemaker_runtime_client, model_name, input_data, target_models, content_type
             )
+
+
+def create_and_invoke_endpoint_helper(
+    image_uri,
+    sagemaker_regions,
+    model_name,
+    model_data,
+    instance_type,
+    accelerator_type,
+    input_data,
+    is_multi_model_mode_enabled=False,
+    target_models=[],
+    environment={},
+    content_type="application/json",
+):
+    ecr_image_region = get_ecr_image_region(image_uri)
+    for region in sagemaker_regions:
+        sagemaker_client = get_sagemaker_client(region)
+        boto_session = boto3.Session(region_name=region)
+        sagemaker_runtime_client = get_sagemaker_runtime_client(region)
+        try:
+            # Reupload the image to test region if needed
+            tested_ecr_image = (
+                get_ecr_image(image_uri, region) if region != ecr_image_region else image_uri
+            )
+            # update model uri to use correct region
+            tested_model_data = get_model_data(model_data, region)
```

Comment:
why are we generating the model uri first with default region and later replacing?

Why cant we use the `util.find_or_put_model_data` function here directly?

Reply:
@tejaschumbalkar so do you want `test_tfs_neuronx_model` and `test_tfs_neuron_model` to not use
```
@pytest.fixture(scope="session")
def tfs_neuronx_model(region, boto_session):
    return util.find_or_put_model_data(region, boto_session, "data/tfs-neuronx-model.tar.gz")
   ```
   as input? You want the path "data/tfs-neuronx-model.tar.gz" to be passed to `create_and_invoke_endpoint_helper` and inside the `for region in sagemaker_regions:` call `find_or_put_model_data` with the current region and model path?